File: python/tests/smoke_test.py
"""
Simple smoke test for DiffRhythm service
"""

import sys
import os

# Add services directory to path
sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..', 'services'))

def test_engine_creation():
    """Test that engine can be created"""
    try:
        from diffrhythm_service import DiffRhythmEngine
        engine = DiffRhythmEngine()
        assert engine is not None
        assert hasattr(engine, 'device')
        assert hasattr(engine, 'model_cache_dir')
        print("✅ DiffRhythmEngine creation test passed")
        return True
    except Exception as e:
        print(f"❌ DiffRhythmEngine creation test failed: {e}")
        return False

def test_job_store_operations():
    """Test basic job store operations"""
    try:
        from diffrhythm_service import JobStore, GenerationRequest
        
        store = JobStore()
        
        # Create job
        request = GenerationRequest(
            prompt="Test prompt",
            durationSeconds=30,
            language="en"
        )
        
        job_id = store.create_job(request)
        assert job_id is not None
        
        # Get job
        job = store.get_job(job_id)
        assert job is not None
        assert job.request_data == request
        assert job.status == "pending"
        
        # Update job
        store.update_job(job_id, progress=50, message="Processing")
        updated_job = store.get_job(job_id)
        assert updated_job.progress == 50
        assert updated_job.message == "Processing"
        
        print("✅ Job store operations test passed")
        return True
    except Exception as e:
        print(f"❌ Job store operations test failed: {e}")
        return False

if __name__ == "__main__":
    print("🚀 Running Python DiffRhythm smoke tests...\n")
    
    tests_passed = 0
    total_tests = 2
    
    if test_engine_creation():
        tests_passed += 1
    
    if test_job_store_operations():
        tests_passed += 1
    
    print(f"\n📊 Test Results: {tests_passed}/{total_tests} tests passed")
    
    if tests_passed == total_tests:
        print("🎉 All smoke tests passed!")
        sys.exit(0)
    else:
        print("💥 Some tests failed!")
        sys.exit(1)

File: backend/tests/smoke_test.js
/**
 * Simple smoke test for backend services
 */

import Database from '../models/Database.js';
import DiffRhythmJobService from '../services/DiffRhythmJobService.js';

async function testDatabase() {
    console.log('🔍 Testing database connection...');
    
    const db = new Database(':memory:'); // Use in-memory database for testing
    await db.connect();
    await db.initialize();
    
    // Test basic operations
    await db.run('INSERT INTO jobs (id, type, status) VALUES (?, ?, ?)', ['test-1', 'test', 'pending']);
    const job = await db.get('SELECT * FROM jobs WHERE id = ?', ['test-1']);
    
    if (job && job.id === 'test-1') {
        console.log('✅ Database operations working');
    } else {
        throw new Error('Database operations failed');
    }
    
    await db.close();
}

async function testJobService() {
    console.log('🔍 Testing DiffRhythm job service...');
    
    const service = new DiffRhythmJobService();
    
    // Mock the database to avoid actual DB operations
    service.db = {
        run: async () => {},
        get: async () => ({ request_data: '{}' }),
        all: async () => [],
        connect: async () => {},
        initialize: async () => {},
        close: async () => {}
    };
    
    // Test job creation validation
    try {
        await service.createJob({
            prompt: '',
            durationSeconds: 30,
            language: 'en'
        });
        throw new Error('Should have failed on empty prompt');
    } catch (error) {
        if (error.message.includes('required')) {
            console.log('✅ Job validation working');
        } else {
            throw error;
        }
    }
    
    try {
        await service.createJob({
            prompt: 'Test prompt',
            durationSeconds: 5, // Too short
            language: 'en'
        });
        throw new Error('Should have failed on duration');
    } catch (error) {
        if (error.message.includes('between 10 and 300')) {
            console.log('✅ Duration validation working');
        } else {
            throw error;
        }
    }
    
    try {
        await service.createJob({
            prompt: 'Test prompt',
            durationSeconds: 30,
            language: 'de' // Invalid language
        });
        throw new Error('Should have failed on language');
    } catch (error) {
        if (error.message.includes('ru or en')) {
            console.log('✅ Language validation working');
        } else {
            throw error;
        }
    }
}

async function runSmokeTests() {
    console.log('🚀 Running backend smoke tests...\n');
    
    try {
        await testDatabase();
        await testJobService();
        console.log('\n✅ All backend smoke tests passed!');
    } catch (error) {
        console.error('\n❌ Smoke test failed:', error.message);
        process.exit(1);
    }
}

// Run tests if this file is executed directly
if (import.meta.url === `file://${process.argv[1]}`) {
    runSmokeTests();
}

export { runSmokeTests };